src/reflect/scala/reflect/internal/Scopes.scala

```diff
@@ -282,6 +282,30 @@ trait Scopes extends api.Scopes { self: SymbolTable =>
       }
     }
 
+    final def lookupSymbolEntry(sym: Symbol): ScopeEntry = {
+      var e = lookupEntry(sym.name)
+      while (e ne null) {
+        if (e.sym == sym) return e
+        e = lookupNextEntry(e)
+      }
+      null
+    }
+
+    final def lookupCompanion(original: Symbol): Symbol = {
+      lookupSymbolEntry(original) match {
+        case null =>
+        case entry if entry.sym == original =>
+          entry.owner.lookupAllEntries(original.name.companionName)
+            .filter(_.owner eq entry.owner)
+            .filter(x => original.isTerm || x.sym.hasModuleFlag).toList match {
```

Comment:
A comment on this filter would help - i guess `original` is either a class symbol or a module symbol? I see it was copied from the original code, but it's not very clear. Also, the original code had a check for `isCoDefinedWith` - is this no longer needed?